Sources/MapboxMaps/Annotations/AnnotationOrchestrator.swift

```diff
@@ -47,34 +52,15 @@ public protocol AnnotationInteractionDelegate: AnyObject {
 }
 
 /// `AnnotationOrchestrator` provides a way to create annotation managers of different types.
-public class AnnotationOrchestrator {
-
-    private let gestureRecognizer: UIGestureRecognizer
-
-    private let style: Style
-
-    private let mapFeatureQueryable: MapFeatureQueryable
-
-    private weak var displayLinkCoordinator: DisplayLinkCoordinator?
+public final class AnnotationOrchestrator {
+    private let impl: AnnotationOrchestratorImplProtocol
 
-    internal init(gestureRecognizer: UIGestureRecognizer,
-                  mapFeatureQueryable: MapFeatureQueryable,
-                  style: Style,
-                  displayLinkCoordinator: DisplayLinkCoordinator) {
-        self.gestureRecognizer = gestureRecognizer
-        self.mapFeatureQueryable = mapFeatureQueryable
-        self.style = style
-        self.displayLinkCoordinator = displayLinkCoordinator
-
-        gestureRecognizer.addTarget(self, action: #selector(handleTap(_:)))
+    internal init(impl: AnnotationOrchestratorImplProtocol) {
+        self.impl = impl
```

Comment:
Why do we need a private implementation with a public facade?

Reply:
I did that to enable annotation orchestrator to be mocked